asset/file_reader/index.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+const Promise = require('bluebird');
+const Queue = require('@terascope/queue');
+const { getChunk, getOffsets } = require('@terascope/chunked-file-reader');
+const fse = require('fs-extra');
+
+function newSlicer(context, executionContext, retryData, logger) {
+    const opConfig = executionContext.config.operations[0];
+    const queue = new Queue();
+
+    async function processFile(filePath) {
+        const stat = await fse.stat(filePath);
+        const total = stat.size;
+        getOffsets(opConfig.size, total, opConfig.delimiter).forEach((offset) => {
+            offset.path = filePath;
+            queue.enqueue(offset);
+        });
+    }
+
+    async function getFilePaths(filePath) {
+        const dirContents = await fse.readdir(filePath);
+        return Promise.map(dirContents, async (file) => {
+            const fullPath = `${filePath}/${file}`;
+            const stats = await fse.stat(fullPath);
```

Comment:
We need to guard against following sym-links.  Looks like we should be using `lstat()`.  On my mac where /tmp is a sym-link:
```
> require('fs-extra').statSync('/tmp').isDirectory()
true
> require('fs-extra').statSync('/tmp').isSymbolicLink()
false
> require('fs-extra').lstatSync('/tmp').isDirectory()
false
> require('fs-extra').lstatSync('/tmp').isSymbolicLink()
true
```

Reply:
Swapped with `lstat` to guard against symlinks